jans-auth-server/model/src/main/java/io/jans/as/model/jws/EDDSASigner.java

```diff
@@ -0,0 +1,137 @@
+
+    /**
+     * Generating a signature,
+     * using URL safe based format.
+     */
+    @Override
+    public String generateSignature(String signingInput) throws SignatureException {
+        SignatureAlgorithm signatureAlgorithm = getSignatureAlgorithm();
+        if (signatureAlgorithm == null) {
+            throw new SignatureException("The signature algorithm is null");
+        }
+        if (!signatureAlgorithm.getFamily().equals(AlgorithmFamily.ED)) {
+            throw new SignatureException(String.format("Wrong value of the signature algorithm: %s", signatureAlgorithm.getFamily().toString()));
+        }
+        if (eddsaPrivateKey == null) {
+            throw new SignatureException("The EDDSA private key is null");
+        }
+        if (signingInput == null) {
+            throw new SignatureException("The signing input is null");
+        }
+        try {
+            PKCS8EncodedKeySpec privateKeySpec = eddsaPrivateKey.getPrivateKeySpec();
+            java.security.KeyFactory keyFactory = java.security.KeyFactory.getInstance(signatureAlgorithm.getName());
+            BCEdDSAPrivateKey privateKey = (BCEdDSAPrivateKey) keyFactory.generatePrivate(privateKeySpec);
+            Signature signer = Signature.getInstance(signatureAlgorithm.getName(), DEF_BC);
+            signer.initSign(privateKey);
+            signer.update(signingInput.getBytes());
+            byte[] signature = signer.sign();
+            return Base64Util.base64urlencode(signature);
+        } catch (NoSuchAlgorithmException | NoSuchProviderException | InvalidKeySpecException | InvalidKeyException
+                | IllegalArgumentException e) {
+            throw new SignatureException(e);
+        }
+    }
+
+    /**
+     * Validating a signature.
+     */
+    @Override
+    public boolean validateSignature(String signingInput, String signature) throws SignatureException {
+        SignatureAlgorithm signatureAlgorithm = getSignatureAlgorithm();
+        if (signatureAlgorithm == null) {
+            throw new SignatureException("The signature algorithm is null");
+        }
+        if (!signatureAlgorithm.getFamily().equals(AlgorithmFamily.ED)) {
+            throw new SignatureException(String.format("Wrong value of the signature algorithm: %s", signatureAlgorithm.getFamily().toString()));
+        }
+        if (eddsaPublicKey == null) {
+            throw new SignatureException("The EDDSA public key is null");
+        }
+        if (signingInput == null) {
+            throw new SignatureException("The signing input is null");
+        }
+        try {
+            X509EncodedKeySpec publicKeySpec = eddsaPublicKey.getPublicKeySpec();
+            java.security.KeyFactory keyFactory = java.security.KeyFactory.getInstance(signatureAlgorithm.getName());
+            BCEdDSAPublicKey publicKey = (BCEdDSAPublicKey) keyFactory.generatePublic(publicKeySpec);
+            Signature virifier = Signature.getInstance(signatureAlgorithm.getName(), "BC");
```

Comment:
DEF_BC instead of "BC" ? Lets use everywhere "BC" or otherwise DEF_BC. Mix does not look consistent.